source/Calamari/Behaviors/TargetDiscoveryBehaviour.cs

```diff
@@ -109,6 +114,51 @@ public async Task Execute(RunningDeployment runningDeployment)
             }
         }
 
+        private IEnumerable<IWebApp> ListWebApps(IAzure azureClient)
+        {
+            var policy = CreateAzureQueryRetryPolicy(5, $"listing web apps");
```

Comment:
Just a general consideration with the exponential backoff: there's a potential 62 seconds of wait time during target discovery. 

But, Target Discovery seems like it's very opt-in at the moment, with both the overall Feature Flag, and then needing to add the specific Account variable before it will kick in. 

Sounds okay to me as long as you're okay/aware of that worst-case total wait time.

Reply:
Hmmm yeah and that is on a per call to Azure basis, so if Azure was having a somewhat bad day then we could potentially add a fair bit of time I suppose before something potentially fails completely and we give up. I initially had the max retries at 3 but found tests were still having issues so bumped it up to 5, we could potentially cap the max delay time to 10 seconds potentially so we would go 2, 4, 8, 10, 10.

Reply:
I'll take a look at this in a follow up PR